packages/components/toggle/src/Toggle.tsx

```diff
@@ -48,14 +52,33 @@ const Toggle = ({
     <ToggleContainerStyled disabled={disabled}>
       <ToggleStyled {...props}>
         <InputStyled
-          {...mergedThemeValues}
+          styled={mergedThemeValues as Required<ToggleStyledProps>}
           type="checkbox"
           checked={checked}
           onChange={handleChange}
+          disabled={disabled}
+          reSize={size}
         />
-        <SpanStyled {...mergedThemeValues} />
+        <SpanStyled
+          size={size}
+          disabled={disabled}
+          withIcon={withIcon}
+          styled={mergedThemeValues as Required<ToggleStyledProps>}
+        >
+          <CircleStyled
+            size={size}
+            styled={mergedThemeValues as Required<ToggleStyledProps>}
+            disabled={disabled}
+            withIcon={withIcon}
+            checked={checked}
+          >
+            {withIcon && iconName && (
+              <Icon name={iconName} data-testid={'icon'} />
```

Comment:
I think the icon props should be a child to allow the user to use a different icon or even an image